### PR TITLE
Catch docstring yields

### PR DESCRIPTION
**Pull request by mishig25**
Catch `yields` section in docstrings

### CODE REVIEW

File: src/doc_builder/convert_rst_to_mdx.py [hunk 1]
```diff
@@ -431,8 +431,10 @@ def parse_rst_docstring(docstring):
 
         # Returns section
         elif _re_returns.search(lines[idx]) is not None:
+            # tag is either `return` or `yield`
```
Comment (sgugger) [inline src/doc_builder/convert_rst_to_mdx.py:434]: Could we group the `Raises` block with this? We would just need a map from `tag` to `tag_type` that sends `raises` to `rasiederrors` and `returns`/`yields` to `returnstype`/`yieldstype`.

**Review by sgugger — COMMENTED** [review]
Thanks for working on this. Could you also add a test?

**Comment by mishig25** [comment]
Thanks a lot for the suggestions! I've applied those suggestions; so please feel free to re-review them 😊

File: src/doc_builder/convert_rst_to_mdx.py [hunk 2]
```diff
@@ -441,21 +408,31 @@ def parse_rst_docstring(docstring):
...
+
             lines.insert(idx, f"</{tag}s>\n")
             idx += 1
 
             # Return block finished, we insert the return type if one was specified
-            if return_type is not None:
+            if tag in ["return", "yield"] and return_type is not None:
                 lines[idx - 1] += f"\n<{tag}type>{return_type}</{tag}type>\n"
+            elif raised_errors:
+                # rasied error
```
Comment (sgugger) [inline src/doc_builder/convert_rst_to_mdx.py]:
```suggestion
                # raised error
```

**Review by sgugger — APPROVED** [review]
Thanks!

**Pull request merged**